common/test/acceptance/pages/studio/overview.py

```diff
@@ -1037,19 +1027,48 @@ def policy(self, grading_label):
         ).fulfill()
 
     @property
+    def is_staff_lock_visible(self):
+        """
+        Returns true if the staff lock option is visible, either as a checkbox
+        (section and unit levels) or a radio button (subsection level).
+        """
+        return self.find_css('#staff_lock').visible or self.find_css('input[name=content-visibility]').visible
+
+    def ensure_staff_lock_visible(self):
+        """
+        Ensures the staff lock option is visible, clicking on the advanced tab
+        if needed.
+        """
+        if not self.is_staff_lock_visible:
+            self.find_css(".settings-tab-button[data-tab=advanced]").click()
```

Comment:
I'm concerned this can cause flaky test behavior.  Here, we are not waiting for the advanced tab to be visible and ready before proceeding.  And since the code is written to work for either subsection and non-subsection blocks, it cannot be robust since it doesn't know which exact UI to expect.

How difficult would it be for the caller to pass down which UI it is expecting?  For subsection or for non-subsection?  If you do this, you can have smaller methods for each and eliminate the various `if` statements in the changes in this file.